ember-intl/addon/-private/utils/translations.ts

```diff
@@ -0,0 +1,40 @@
+type NestedStructure<T = string> = {
+  [Key in string | number]?: T | NestedStructure<T>;
+};
+
+export type Translations = NestedStructure<string | number>;
+
+/**
+ * @private
+ * @hide
+ */
+export function flattenKeys<T>(object: NestedStructure<T>): Record<string, T> {
```

Comment:
I renamed the function to better describe what it does.

Sample input:

```ts
{
  "components": {
    "macros-classic": {
      "hello": {
        "message": "Hello, {name}!"
      }
    }
  },
  "routes": {
    "application": {
      "title": "ember-intl"
    },
    "smoke-tests": {
      "title": "Smoke tests"
    }
  },
  "smoke-tests": {
    "parent": {
      "child": "Hello world!"
    }
  }
}
```

Sample output:

```ts
{
  "components.macros-classic.hello.message": "Hello, {name}!",
  "routes.application.title": "ember-intl",
  "routes.smoke-tests.title": "Smoke tests",
  "smoke-tests.parent.child": "Hello world!"
}
```